src/util/psbt/mod.rs

```diff
@@ -88,6 +91,82 @@ impl PartiallySignedTransaction {
 
         Ok(())
     }
+
+    /// Decide whether the current Input spends segwit outputs or not and
+    /// calculate the corresponding Sighash.
```

Comment:
This comment should be changed to indicate that it looks at the PSBT data to determine the appropriate sighash mode as well as whether to sign using segwit or legacy. As written it sounds like its primary job is to decide between segwit/non-segwit which isn't true.